src/api/mixPanel.ts

```diff
@@ -0,0 +1,86 @@
+import { getExternalId } from "$/auth/getAuthHeader";
+import { Agent } from "..";
+
+export interface AnalyticsOptions {
+    mixPanelKey: string;
+    agent: Agent;
+    chatId?: string;
+    isEnabled?: boolean;
+    distinctId?: string;
+}
+
+class AnalyticsProvider {
+    private static instance: AnalyticsProvider;
+    private mixPanelKey: string;
+    private distinct_id?: string;
+    private isEnabled: boolean = true;
+    private chatId?: string;
+    private agentId: string;
+    private owner_id: string;
+
+    private constructor(config: AnalyticsOptions) {
+        this.mixPanelKey = config.mixPanelKey || 'testKey';
+        this.distinct_id = config.distinctId || getExternalId();
+        this.isEnabled = config.isEnabled || true;
+        this.chatId = config.chatId;
+        this.agentId = config.agent.id;
+        this.owner_id = config.agent.owner_id ?? ''
+    }
+
+    static getInstance(config?: AnalyticsOptions) {
```

Comment:
make the instance module level and export it, don't pass the analytics around functions